src/VisualStudio/Core/Def/Packaging/PackageSearchService.IOService.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.VisualStudio.LanguageServices.Packaging
+{
+    internal partial class PackageSearchService
+    {
+        private class IOService : IPackageSearchIOService
+        {
+            public void Create(DirectoryInfo directory) => directory.Create();
+
+            public void Delete(FileInfo file) => file.Delete();
+
+            public bool Exists(FileSystemInfo info) => info.Exists;
+
+            public byte[] ReadAllBytes(string path) => File.ReadAllBytes(path);
+
+            public void Replace(string sourceFileName, string destinationFileName, string destinationBackupFileName, bool ignoreMetadataErrors) =>
+                File.Replace(sourceFileName, destinationFileName, destinationBackupFileName, ignoreMetadataErrors);
+
+            public void Move(string sourceFileName, string destinationFileName) =>
+                File.Move(sourceFileName, destinationFileName);
+
+            public void WriteAndFlushAllBytes(string path, byte[] bytes)
+            {
+                using (var fileStream = new FileStream(path, FileMode.Create))
+                {
+                    fileStream.Write(bytes, 0, bytes.Length);
+                    fileStream.Flush(flushToDisk: true);
```

Comment:
It's not doc'ed if Dispose will call "flushToDisk: true" or not though.  So this is me being paranoid in case they don't.